posthog/api/test/test_feature_flag.py

```diff
@@ -150,3 +150,16 @@ def test_cannot_delete_feature_flag_on_another_team(self, mock_capture):
         self.assertTrue(FeatureFlag.objects.filter(pk=self.feature_flag.pk).exists())
 
         mock_capture.assert_not_called()
+
+    def test_creating_a_feature_flag_with_same_team_and_key_after_deleting(self):
+        instance = FeatureFlag.objects.create(team=self.team, created_by=self.user, key="alpha-feature")
+        instance.delete()
```

Comment:
This doesn't test the right thing. You can just pass `deleted=True` to the line above.